src/stringify/stringifyString.js

```diff
@@ -219,16 +219,23 @@ function plainString(item, ctx, onComment, onChompKeep) {
       value
     )
   ) {
+    // prettier-ignore
+    const defaultQuotedString =
+      value.indexOf('"') !== -1 && value.indexOf("'") === -1
+        ? singleQuotedString
+        : value.indexOf("'") !== -1 && value.indexOf('"') === -1
+          ? doubleQuotedString
+          : defaultQuote === Type.QUOTE_SINGLE
+            ? singleQuotedString
+            : doubleQuotedString
     // not allowed:
     // - empty string, '-' or '?'
     // - start with an indicator character (except [?:-]) or /[?-] /
     // - '\n ', ': ' or ' \n' anywhere
     // - '#' not preceded by a non-space char
     // - end with ' ' or ':'
     return implicitKey || inFlow || value.indexOf('\n') === -1
-      ? value.indexOf('"') !== -1 && value.indexOf("'") === -1
-        ? singleQuotedString(value, ctx)
-        : doubleQuotedString(value, ctx)
+      ? defaultQuotedString(value, ctx)
       : blockString(item, ctx, onComment, onChompKeep)
```

Comment:
With this added complexity, the ternaries should be dropped and an `if` statement used instead. Wanting to add a `prettier-ignore` comment is rather indicative of this.

This should help avoid useless computations by moving them to only the path where they're used, as well as not indexOf'ing value multiple times. Also, `defaultQuotedString` should probably rather be called `quotedString`.

Reply:
OK, I expanded it into if/else chain, but I really not sure it became better